### PR TITLE
[@types/apidoc] Add apidoc typings

### PR DESCRIPTION
**Pull request by rigwild**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

___

See https://github.com/apidoc/apidoc/issues/875
Ping @NicolasCARPi

### CODE REVIEW

**Comment by rigwild** [comment]
I don't know how to fix the error. The package does not have a default export.

```
Module '"/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/apidoc/index"' has no default export.
```

I tried:

```ts
export default { createDoc }
export = { createDoc }
export = {}
export {}
```

File: types/apidoc/apidoc-tests.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import apidoc from 'apidoc';
+
+const apidocOutput = apidoc({
```
Comment (demensky) [inline types/apidoc/apidoc-tests.ts:3]: > I don't know how to fix the error. The package does not have a default export.

The library really does not have default export:

```js
var apidoc = require("apidoc")

console.log(apidoc); // Object {createDoc: function()}
```

```suggestion
import {createDoc} from 'apidoc';

const apidocOutput = createDoc({
```

**Review by demensky — CHANGES_REQUESTED** [review]

File: types/apidoc/apidoc-tests.ts (hunk 1 above)
Comment (rigwild) [reply types/apidoc/apidoc-tests.ts]: Looks like it worked. Thanks!

**Review by demensky — APPROVED** [review]
I could not find the documentation for this function. But I think that for the first time this definition should be enough.

In the future, it makes sense to make definitions for `apidoc-core` since [`options` are passed to it](https://github.com/apidoc/apidoc/blob/6af3f459af4c72b904e5776344fcd3334b6531ea/lib/index.js#L130).

**Pull request merged**